pkg/metrics/metrics.go

```diff
@@ -276,14 +281,8 @@ func (c *Counter) UpdateMetricValidationSucceeded(upgradeConfigName string) {
 		float64(0))
 }
 
-func (c *Counter) UpdateMetricClusterCheckFailed(upgradeConfigName string) {
-	metricClusterCheckFailed.With(prometheus.Labels{
-		nameLabel: upgradeConfigName}).Set(
-		float64(1))
-}
-
-func (c *Counter) UpdateMetricClusterCheckSucceeded(upgradeConfigName string) {
-	metricClusterCheckFailed.With(prometheus.Labels{
+func (c *Counter) UpdateMetricHealthcheckSucceeded(upgradeConfigName string) {
+	metricHealthcheckFailed.With(prometheus.Labels{
```

Comment:
`metricHealthCheckFailed` accepts two labels `[]string{nameLabel, failedReason})` however for success we are only passing one which causes below panic:
```
ts=2024-05-15T04:27:02.21703987Z level=info logger=controller_upgradeconfig msg="Prehealth check for critical alerts passed" Request.Namespace=openshift-managed-upgrade-operator Request.Name=managed-upgrade-config
ts=2024-05-15T04:27:02.240432433Z level=info logger=controller_upgradeconfig msg="Prehealth check for clusteroperators passed" Request.Namespace=openshift-managed-upgrade-operator Request.Name=managed-upgrade-config
ts=2024-05-15T04:27:02.240471445Z level=info msg="Observed a panic in reconciler: inconsistent label cardinality: expected 2 label values but got 1 in prometheus.Labels{\"upgradeconfig_name\":\"managed-upgrade-config\"}" controller=upgradeconfig controllerGroup=upgrade.managed.openshift.io controllerKind=UpgradeConfig UpgradeConfig="{managed-upgrade-config openshift-managed-upgrade-operator}" namespace=openshift-managed-upgrade-operator name=managed-upgrade-config reconcileID=00ac51aa-e9a7-4d8a-9e51-d6b57a1a77fa
panic: inconsistent label cardinality: expected 2 label values but got 1 in prometheus.Labels{"upgradeconfig_name":"managed-upgrade-config"} [recovered]
        panic: inconsistent label cardinality: expected 2 label values but got 1 in prometheus.Labels{"upgradeconfig_name":"managed-upgrade-config"}

goroutine 196 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        pkg/mod/sigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x192c9c0?, 0xc00030f0e0?})
        /usr/lib/golang/src/runtime/panic.go:914 +0x21f
```

The following will fix it:
- We can either have `metricHealthCheckSucceeded` gauge var separately and just set the upgradeConfigName
- Or we can continue to use `metricHealthcheckFailed` but with `failedReason: "none"` 

I think it's better to have separate `metricHealthCheckSucceeded` var to avoid confusion and not tie to what labels the healthcheck failure has. Can you add this fix?

Reply:
Thanks, @ravitri for testing.
I have address this in my latest commit